src/gt4py/backend/templates/stencil_module.py.in

```diff
@@ -135,15 +143,57 @@ class {{ class_name }}(StencilObject):
             domain=domain,
             origin=origin,
             validate_args=validate_args,
-            exec_info=exec_info,
+            exec_info=stencil_info,
         )
 
 {%- filter indent(width=8) %}
 {{ post_run }}
 {%- endfilter %}
 
-        if exec_info is not None:
-            exec_info["call_end_time"] = time.perf_counter()
+        if stencil_info is not None:
+            stencil_info["call_end_time"] = time.perf_counter()
+
+            # compute statistics
+            stencil_info["call_time"] = (
+                stencil_info["call_end_time"]
+                - stencil_info["call_start_time"]
+            )
+            stencil_info["total_call_time"] = (
+                stencil_info.get("total_call_time", 0.0)
+                + stencil_info["call_time"]
+            )
+            stencil_info["ncalls"] = (
+                stencil_info.get("ncalls", 0) + 1
+            )
+            stencil_info["average_call_time"] = (
+                stencil_info["total_call_time"]
+                / stencil_info["ncalls"]
+            )
+            stencil_info["run_time"] = (
+                stencil_info["run_end_time"]
+                - stencil_info["run_start_time"]
+            )
+            stencil_info["total_run_time"] = (
+                stencil_info.get("total_run_time", 0.0)
+                + stencil_info["run_time"]
+            )
+            stencil_info["average_run_time"] = (
+                stencil_info["total_run_time"]
+                / stencil_info["ncalls"]
+            )
+            if "run_cpp_start_time" in stencil_info:
+                stencil_info["run_cpp_time"] = (
+                    stencil_info["run_cpp_end_time"]
+                    - stencil_info["run_cpp_start_time"]
+                )
+                stencil_info["total_run_cpp_time"] = (
+                    stencil_info.get("total_run_cpp_time", 0.0)
+                    + stencil_info["run_cpp_time"]
+                )
+                stencil_info["average_run_cpp_time"] = (
+                    stencil_info["total_run_cpp_time"]
+                    / stencil_info["ncalls"]
+                )
```

Comment:
This part can also be updated in a way that is compatible with the current `exec_info` usage. I would suggest to drop the computation of the averages, since it can be trivially done outside of the stencil call without adding further overhead.

```suggestion
        if exec_info is not None:
            exec_info["call_end_time"] = stencil_info["call_end_time"] = time.perf_counter()       

            # Update performance counters
            stencil_info["num_calls"] = (
                stencil_info.get("num_calls", 0) + 1
            )
            stencil_info["call_time"] = (
                stencil_info["call_end_time"]
                - stencil_info["call_start_time"]
            )
            stencil_info["total_call_time"] = (
                stencil_info.get("total_call_time", 0.0)
                + stencil_info["call_time"]
            )
            stencil_info["run_time"] = (
                stencil_info["run_end_time"]
                - stencil_info["run_start_time"]
            )
            stencil_info["total_run_time"] = (
                stencil_info.get("total_run_time", 0.0)
                + stencil_info["run_time"]
            )

            if "run_cpp_start_time" in stencil_info:
                stencil_info["run_cpp_time"] = (
                    stencil_info["run_cpp_end_time"]
                    - stencil_info["run_cpp_start_time"]
                )
                stencil_info["total_run_cpp_time"] = (
                    stencil_info.get("total_run_cpp_time", 0.0)
                    + stencil_info["run_cpp_time"]
                )
```